components/observatorium.libsonnet

```diff
@@ -185,19 +185,29 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
     },
   },
 
-  queryCache:: (import 'cortex-query-frontend.libsonnet') + {
-    config+:: {
-      local cfg = self,
-      name: obs.config.name + '-' + cfg.commonLabels['app.kubernetes.io/name'],
-      namespace: obs.config.namespace,
-      commonLabels+:: obs.config.commonLabels,
-      downstreamURL: 'http://%s.%s.svc.cluster.local.:%d' % [
-        obs.query.service.metadata.name,
-        obs.query.service.metadata.namespace,
-        obs.query.service.spec.ports[1].port,
-      ],
+  queryFrontend::
+    t.queryFrontend +
+    t.queryFrontend.withServiceMonitor +
+    t.queryFrontend.withSplitInterval +
+    t.queryFrontend.withMaxRetries +
+    t.queryFrontend.withLogQueriesLongerThan +
+    t.queryFrontend.withInMemoryResponseCache + {
+      config+:: {
+        local cfg = self,
+        name: obs.config.name + '-' + cfg.commonLabels['app.kubernetes.io/name'],
+        namespace: obs.config.namespace,
+        commonLabels+:: obs.config.commonLabels,
+        replicas: 1,
+        downstreamURL: 'http://%s.%s.svc.cluster.local.:%d' % [
+          obs.query.service.metadata.name,
+          obs.query.service.metadata.namespace,
+          9090,
```

Comment:
Sorry, I just found I hardcode the port. It should be `obs.query.service.spec.ports[1].port`. I will open another pr to fix it. @kakkoyun